### PR TITLE
Darwinia <> Darwinia Parachain

### PR DESCRIPTION
**Pull request by acgxv**

### CODE REVIEW

File: runtime/darwinia-parachain/src/pallets/helixbridge.rs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+use crate::*;
+use bp_message_dispatch::CallOrigin;
+use bp_messages::{LaneId, MessageNonce};
+use bp_runtime::{messages::DispatchFeePayment, ChainId, DARWINIA_CHAIN_ID};
+use bridge_runtime_common::lanes::DARWINIA_DARWINIA_PARACHAIN_LANE;
+use bridges_message::darwinia::ToDarwiniaMessagePayload;
+use dp_common_runtime::helixbridge::{CallParams, Config, CreatePayload, LatestMessageNoncer};
+use pallet_bridge_messages::{outbound_lane, Instance1 as WithDarwiniaMessages};
+
+/// The s2s backing pallet index in the darwinia chain runtime.
```
Comment (boundless-forest) [inline runtime/darwinia-parachain/src/pallets/helixbridge.rs]: Please update this comment.

File: runtime/darwinia-parachain/src/pallets/helixbridge.rs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+const DARWINIA_S2S_BACKING_PALLET_INDEX: u8 = 57;
```
Comment (boundless-forest) [inline runtime/darwinia-parachain/src/pallets/helixbridge.rs]: @xiaoch05 Please help review this. `57` refers to the fee-market in the latest darwinia runtime. So I think this is incorrect here.

https://github.com/darwinia-network/darwinia/blob/main/runtime/darwinia/src/lib.rs#L259

Comment (xiaoch05) [reply runtime/darwinia-parachain/src/pallets/helixbridge.rs]: yes, I will check all the configure variable for this pallet.

File: runtime/darwinia-parachain/src/pallets/remote_governance.rs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+// --- paritytech ---
```
Comment (boundless-forest) [inline runtime/darwinia-parachain/src/pallets/remote_governance.rs:1]: @jiguantong @HackFisher Please help check the config for this pallet.

Comment (jiguantong) [reply runtime/darwinia-parachain/src/pallets/remote_governance.rs:1]: looks ok

File: runtime/darwinia-parachain/src/pallets/remote_governance.rs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	for EnsureSpecific
+{
+	type Success = AccountId;
+
+	fn try_origin(o: O) -> Result<Self::Success, O> {
+		o.into().and_then(|o| match o {
+			RawOrigin::Signed(who)
+				if who
+					== array_bytes::hex_n_into_unchecked(
+						"0xc778fc2665f3f6ee9623594e5d1fab9dbd557149542c5edacbcc543a82c9d780",
```
Comment (hackfisher) [inline runtime/darwinia-parachain/src/pallets/remote_governance.rs:19]: How this derive account generated?

Is chain id included in derive process?

**Review by hackfisher — COMMENTED** [review]
@AurevoirXavier CI failed

File: runtime/darwinia-parachain/src/pallets/remote_governance.rs (hunk 4 above)
Comment (acgxv) [reply runtime/darwinia-parachain/src/pallets/remote_governance.rs:19]: Previous multi-sig root key.

Comment (hackfisher) [reply runtime/darwinia-parachain/src/pallets/remote_governance.rs:19]: "multi-sig root key" looks not correct, this should be the LCMP derived key from Darwinia's Root?

Comment (hackfisher) [reply runtime/darwinia-parachain/src/pallets/remote_governance.rs:19]: It looks the same as Crab's relevant key: https://github.com/darwinia-network/darwinia-parachain/blob/315888f9f024004ab54cad67f4ce6321cda6ae3e/runtime/crab-parachain/src/pallets/remote_governance.rs#L19

Comment (hackfisher) [reply runtime/darwinia-parachain/src/pallets/remote_governance.rs:19]:
```
pub fn derive_account_id<AccountId>(bridge_id: ChainId, id: SourceAccount<AccountId>) -> H256
where
	AccountId: Encode,
{
	match id {
		SourceAccount::Root =>
			(ROOT_ACCOUNT_DERIVATION_PREFIX, bridge_id).using_encoded(blake2_256),
		SourceAccount::Account(id) =>
			(ACCOUNT_DERIVATION_PREFIX, bridge_id, id).using_encoded(blake2_256),
	}
	.into()
}
```

This is the derive algorithm, different chains for same origin(ROOT) will result different derived account.

Please add comments here too.

**Review by hackfisher — CHANGES_REQUESTED** [review]
Fix `EnsureSpecific`

File: runtime/darwinia-parachain/src/pallets/remote_governance.rs (hunk 4 above)
Comment (acgxv) [reply runtime/darwinia-parachain/src/pallets/remote_governance.rs:19]: image
```
λ subalfred key 5GaFKsnKnc2gKQwrCBJzfMxBrS2A7rjrLC4qPynJzav3SJCu
public-key 0xc778fc2665f3f6ee9623594e5d1fab9dbd557149542c5edacbcc543a82c9d780
Substrate 5GaFKsnKnc2gKQwrCBJzfMxBrS2A7rjrLC4qPynJzav3SJCu
```

Comment (acgxv) [reply runtime/darwinia-parachain/src/pallets/remote_governance.rs:19]: I'm pretty sure. It's multi-sig root key.

Comment (hackfisher) [reply runtime/darwinia-parachain/src/pallets/remote_governance.rs:19]: I see, my mistake, I thought it was:

https://github.com/darwinia-network/darwinia-parachain/blob/main/runtime/common/src/remote_governance/mod.rs#L201-L206

**Review by hackfisher — APPROVED** [review]

**Pull request merged**